src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java

```diff
@@ -131,6 +131,15 @@ void crossrefAndInAuthorNames() {
                         database), DEFAULT_UNWANTED_CHARACTERS));
     }
 
+    @Test
+    void latexCommandsAreStrippedFromCitationKey() throws ParseException {
+        BibEntry entry = new BibEntry()
+                .withField(StandardField.TITLE, "Building \\mkbibquote{Community}");
+        String pattern = "[bibtexkey] - [fulltitle]";
```

Comment:
Now I understand where your hacks with `trim()` and removal of `-` come from. However, you better use `.withCitatoinKey("demo24")` after `new BibEntry()` to set an citation key.